### PR TITLE
Version 0.22.0: allow turn off autodetect on server

### PR DESCRIPTION
**Pull request by kenan-gillet**

### CODE REVIEW

File: include/ox_thrift.hrl [hunk 1]
```diff
@@ -15,9 +15,10 @@
           service_module :: atom(),
           %% The protocol module (ox_thrift_protocol_binary or
           %% ox_thrift_protocol_compact) that specifies the default protocol
-          %% that is used to decode and encode messages.  ox_thrift_server now
-          %% automatically detects the protocol used in the incoming request
-          %% and will use that protocol to encode the reply.
+          %% that is used to decode and encode messages.
+          %% ox_thrift_server can automatically detects the protocol used in
+          %% the incoming request and will use that protocol to encode the reply
+          %% if unset ie set to undefined
```
Comment (michal-strus) [inline include/ox_thrift.hrl:21]: That's my suggestion for it, but look at the other comment.
```suggestion
          %% that is used to decode and encode messages.
          %% If unset (i.e. set to undefined) ox_thrift_server will decide between
          %% the two defaulting protocols to encode the reply automatically
          %% based on incoming request. 
```

File: src/ox_thrift_server.erl [hunk 2]
```diff
@@ -245,10 +245,10 @@ encode (#ts_config{service_module=ServiceModule, stats_module=StatsModule}, Prot
...
 
 -spec detect_protocol(Buffer::binary(), Default::atom()) -> atom().
-detect_protocol (<<16#82, _/binary>>, _Default)       -> ox_thrift_protocol_compact;
-detect_protocol (<<16#80, 16#1, _/binary>>, _Default) -> ox_thrift_protocol_binary; %% binary version 1
-detect_protocol (<<0, 0, _/binary>>, _Default)        -> ox_thrift_protocol_binary; %% binary version 0
-detect_protocol (_, Default)                          -> Default.
+detect_protocol (_, Protocol) when Protocol =/= undefined -> Protocol;
+detect_protocol (<<16#82, _/binary>>, undefined)          -> ox_thrift_protocol_compact;
+detect_protocol (<<16#80, 16#1, _/binary>>, undefined)    -> ox_thrift_protocol_binary; %% binary version 1
+detect_protocol (<<0, 0, _/binary>>, undefined)           -> ox_thrift_protocol_binary. %% binary version 0
```
Comment (michal-strus) [inline src/ox_thrift_server.erl:251]: I'm not sure, if that will do what you want? After changes:
* when you specify `Protocol`, then that `Protocol` will be used **regardless** of `Buffer` decoding;
* you have to **unset** `Protocol` for it to be auto-deteced between the two.

Just want to make sure it's in line with others comments, PR desc and intentions.

Comment (kenan-gillet) [reply src/ox_thrift_server.erl:251]: yep, before the PR, Default was never used, so setting protocol_module is kind of useless
I just wanted a way to force use another module like in https://github.com/openx/delivery-market-opportunity-server/pull/724

Comment (michal-strus) [reply src/ox_thrift_server.erl:251]: Yep, I was also missing example usage :) now I see

**Review by michal-strus — APPROVED** [review]

**Pull request merged**